drivers/pwm/pwm_nrfx.c

```diff
@@ -407,73 +415,25 @@ static int pwm_nrfx_pm_control(struct device *dev,
 	};								      \
 	PWM_NRFX_PM_CONTROL(idx)					      \
 	DEVICE_DEFINE(pwm_nrfx_##idx,					      \
-		      DT_NORDIC_NRF_PWM_PWM_##idx##_LABEL,		      \
+		      DT_LABEL(PWM(idx)),				      \
 		      pwm_nrfx_init, pwm_##idx##_nrfx_pm_control,	      \
 		      &pwm_nrfx_##idx##_data,				      \
 		      &pwm_nrfx_##idx##config,				      \
 		      POST_KERNEL, CONFIG_KERNEL_INIT_PRIORITY_DEVICE,	      \
 		      &pwm_nrfx_drv_api_funcs)
 
 #ifdef CONFIG_PWM_0
```

Comment:
Can we make this based on NODELABEL existing instead of CONFIG_PWM_x

Reply:
Together with this change, also these lines:
https://github.com/zephyrproject-rtos/zephyr/blob/72548b39a873c2b1dce36ad2000de2b3676a3443/drivers/pwm/Kconfig.nrfx#L9-L12
need to be replaced with:
```
select NRFX_PWM0 if $(dt_nodelabel_enabled,pwm0)
...

Reply:
@anangl If we're going to make Kconfig changes, I guess we might as well just move this driver entirely to DT instead, so I'll do that using `def_bool $(dt_compat_enabled,$(DT_COMPAT_NORDIC_NRF_PWM))` to avoid 'stuck' symbols.

Sorry for being a bit inconsistent about that in my PRs, I changed my game plan mid way to stop addressing kconfig for zephyr 2.3 after discussing with @galak.

Reply:
We need to keep them, so that is possible to use nrfx drivers directly, not only through Zephyr driver shims. See https://github.com/zephyrproject-rtos/hal_nordic/pull/35#discussion_r412119176.

Reply:
I was afraid this might happen :).

There is a lot to discuss in hal_nordic #35 and I don't want it to block this PR.

@galak @anangl can we agree to just leave Kconfig as-is in this PR, and defer any changes to future work? I just want to get the API conversion done, as per-instance Kconfigs and how they "ought" to work seems to still be under discussion, and anyway is not for zephyr 2.3.

Reply:
I'm fine with that.